### PR TITLE
Add(route): add Google Fonts and change to V2

### LINKED ISSUE #4229: RSS Request: Google Fonts - New Fonts
Sorry, didn't notice that there is an option for 'RSS Requests'.

### What feature is it?

RSS for new fonts available on [Google Fonts](https://fonts.google.com)

### What problem does this feature solve?

Allow users to learn about newly available fonts on Google Fonts. Good for web designers as Google Fonts is easy to use, whlist offering best optiumisation possible for webfonts without any hassles on the developers' side.

### Additional description

Potential ideas:
 - [https://github.com/google/fonts/pulls?q=is%3Apr+is%3Amerged+add](https://github.com/google/fonts/pulls?q=is%3Apr+is%3Amerged+add), Google Fonts' GitHub repository, filtering 'merged' PRs while containing the keyword 'add'. However this contains version-based updates (so not new fonts all the time).
- [https://fonts.google.com/?sort=date](https://fonts.google.com/?sort=date)
- [Google Fonts API](https://developers.google.com/fonts/docs/developer_api#a_quick_example):
   ```bash
   curl \
  'https://www.googleapis.com/webfonts/v1/webfonts?sort=date&key=[YOUR_API_KEY]' \
  --header 'Accept: application/json' \
  --compressed
   ```

    Example response:
```bash
{
 "kind": "webfonts#webfontList",
 "items": [
  {
   "kind": "webfonts#webfont",
   "family": "Caladea",
   "category": "serif",
   "variants": [
    "regular",
    "italic",
    "700",
    "700italic"
   ],
   "subsets": [
    "latin",
    "latin-ext"
   ],
   "version": "v1",
   "lastModified": "2020-03-03",
   "files": {
    "regular": "http://fonts.gstatic.com/s/caladea/v1/kJEzBugZ7AAjhybUjR93-9IztOc.ttf",
    "italic": "http://fonts.gstatic.com/s/caladea/v1/kJExBugZ7AAjhybUvR19__A2pOdvDA.ttf",
    "700": "http://fonts.gstatic.com/s/caladea/v1/kJE2BugZ7AAjhybUtaNY39oYqO52FZ0.ttf",
    "700italic": "http://fonts.gstatic.com/s/caladea/v1/kJE0BugZ7AAjhybUvR1FQ98SrMxzBZ2lDA.ttf"
   }
  },
  {
   "kind": "webfonts#webfont",
   "family": "Oxanium",
   "category": "display",
   "variants": [
    "200",
    "300",
    "regular",
    "500",
    "600",
    "700",
    "800"
   ],
   "subsets": [
    "latin",
    "latin-ext"
   ],
   "version": "v1",
   "lastModified": "2020-03-03",
   "files": {
    "200": "http://fonts.gstatic.com/s/oxanium/v1/RrQVboN_4yJ0JmiMc63l9Lhqa48pA8w.ttf",
    "300": "http://fonts.gstatic.com/s/oxanium/v1/RrQVboN_4yJ0JmiMc8nm9Lhqa48pA8w.ttf",
    "regular": "http://fonts.gstatic.com/s/oxanium/v1/RrQQboN_4yJ0JmiMS2XO0LBBd4Y.ttf",
    "500": "http://fonts.gstatic.com/s/oxanium/v1/RrQVboN_4yJ0JmiMc5Hn9Lhqa48pA8w.ttf",
    "600": "http://fonts.gstatic.com/s/oxanium/v1/RrQVboN_4yJ0JmiMc73g9Lhqa48pA8w.ttf",
    "700": "http://fonts.gstatic.com/s/oxanium/v1/RrQVboN_4yJ0JmiMc9nh9Lhqa48pA8w.ttf",
    "800": "http://fonts.gstatic.com/s/oxanium/v1/RrQVboN_4yJ0JmiMc8Xi9Lhqa48pA8w.ttf"
   }
  },
```

### PR DESCRIPTION
**Pull request by Fatpandac**
## 该 PR 相关 Issue / Involved issue

Close #4229

## 完整路由地址 / Example for the proposed route(s)

```routes
/google/fonts/date
/google/fonts/trending
/google/fonts/popularity
/google/fonts/alpha
/google/fonts/style
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [X] New Route
- [X] Documentation
  - [X] CN
  - [X] EN
- [ ] 全文获取 fulltext
  - [ ] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [X] 日期和时间 date and time
  - [X] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

## 说明 / Note
需要设置 API key

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -58,6 +58,24 @@ Behance 用户主页 URL 获取用户名，如 <https://www.behance.net/mishapet
...
+
+<Route author="Fatpandac" example="/google/fonts/date" path="/google/fonts/:sort?" :paramsDesc="['排序，见下表，默认为最新']" selfhost="1">
+
+| 最新 |   趋势   |  最受欢迎  |  名字 | 风格数量 |
+| :--: | :------: | :--------: | :---: | :------: |
+| date | trending | popularity | alpha |   style  |
+
+::: warning 注意
+
+需要设置 API key，所以只能自建，详情见[ 部署页面 ](https://docs.rsshub.app/en/install/#configuration-route-specific-configurations)的配置模块。
```
Comment (TonyRL) [inline docs/design.md:73]:
```suggestion
需要设置 API key，所以只能自建，详情见[ 部署页面 ](https://docs.rsshub.app/install/#pei-zhi-bu-fen-rss-mo-kuai-pei-zhi)的配置模块。
```

File: lib/v2/google/maintainer.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+    '/sites/:id': ['hoilc'],
+    '/doodles/:language?': ['xyqfer'],
+    '/album/:id': ['hoilc'],
+    '/scholar/:query': ['HenryQW'],
+    '/citations/:id': ['KellyHwong'],
+    '/fonts/:sort?': ['Fatpandac'],
```
Comment (TonyRL) [inline lib/v2/google/maintainer.js:7]: Please sort the routes in alphabetical order

File: lib/v2/google/router.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module.exports = function (router) {
+    router.get('/citations/:id', require('./citations'));
+    router.get('/scholar/:query', require('./scholar'));
+    router.get('/doodles/:language?', require('./doodles'));
+    router.get('/album/:id', require('./album'));
+    router.get('/sites/:id', require('./sites'));
+    router.get('/fonts/:sort?', require('./fonts'));
```
Comment (TonyRL) [inline lib/v2/google/router.js:7]: Please sort the routes in alphabetical order

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/google/fonts.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+const titleMap = {
+    date: 'Newest',
+    popularity: 'Most Popular',
+    trending: 'Trending',
+    alpha: 'Name',
+    style: 'Number of styles',
+};
+
+module.exports = async (ctx) => {
+    const sort = ctx.params.sort || 'date';
```
Comment (TonyRL) [inline lib/v2/google/fonts.js:16]:
```suggestion
    const sort = ctx.params.sort ?? 'date';
```

**Review by TonyRL — CHANGES_REQUESTED** [review]

**Comment by TonyRL** [comment]
Please provide screenshots of the xmls and feed content in RSS reader(that you preferred)

**Comment by Fatpandac** [comment]
@TonyRL 
image

**Pull request merged**